src/main/java/com/thealgorithms/dynamicprogramming/SubsetSum.java

```diff
@@ -9,28 +9,31 @@ private SubsetSum() {
      *
      * @param arr the array containing integers.
      * @param sum the target sum of the subset.
-     * @return {@code true} if a subset exists that sums to the given value, otherwise {@code false}.
+     * @return {@code true} if a subset exists that sums to the given value,
+     *         otherwise {@code false}.
      */
     public static boolean subsetSum(int[] arr, int sum) {
         int n = arr.length;
-        boolean[][] isSum = new boolean[n + 1][sum + 1];
 
-        // Initialize the first column to true since a sum of 0 can always be achieved with an empty subset.
-        for (int i = 0; i <= n; i++) {
-            isSum[i][0] = true;
-        }
+        // Intialize Two Arrays to store current and prev states
+        boolean[] isSumCurr = new boolean[sum + 1];
+        boolean[] isSumPrev = new boolean[sum + 1];
```

Comment:
While you're already using a more efficient space approach (two 1D arrays), it can be improved further by using only one array. You don't actually need both `isSumCurr `and `isSumPrev`. 

By processing the array in reverse order (starting from sum and decrementing down to 0), you can reuse a single array.